Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -0,0 +1,212 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Shared
+import Storage
+import UIKit
+import WebKit
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+// MARK: - OpenSearch
+
+public struct OpenSearchReference: Codable {
+    let reference: String
+    let title: String?
+
+    private enum CodingKeys: String, CodingKey {
+        case reference = "href"
+        case title = "title"
+    }
+}
+
+// MARK: - OpenSearch Browser Extension
+
+extension BrowserViewController {
+    
+    /// Adding Toolbar button over the keyboard for adding Open Search Engine
+    /// - Parameter webView: webview triggered open seach engine
+    func evaluateWebsiteSupportOpenSearchEngine(_ webView: WKWebView) {
+        let script = """
+                        var link = document.querySelector("link[type='application/opensearchdescription+xml']")
+                        var dict = {
+                            href : link.getAttribute("href"),
+                            title : link.getAttribute("title")
+                        };
+                        JSON.stringify(dict)
```

Comment:
Is this valid JS?

`JSON.stringify(dict)` isn't returning a value from a function or anything.
IE: 

`var link = document.querySelector(.....)`
`var dict = {....}`
`JSON.stringify(dict)` but not assigned to anything or returned anywhere?

Reply:
Yes ofcourse it is valid and it works. In webview we are evaluating the result of [opensearchdecription](https://developer.mozilla.org/en-US/docs/Web/OpenSearch) and getting the reference link and title from it,